bugshaker/src/main/java/com/github/stkent/bugshaker/flow/email/screenshot/maps/MapScreenshotProvider.java

```diff
@@ -112,15 +116,20 @@ public MapScreenshotProvider(
     private List<MapView> locateMapViewsInHierarchy(@NonNull final View view) {
         final List<MapView> result = new ArrayList<>();
 
-        if (view instanceof MapView && view.getVisibility() == View.VISIBLE) {
-            // Yes, MapView is a ViewGroup, but I never want to see anyone nesting a MapView
-            // inside another MapView...
-            result.add((MapView) view);
-        } else if (view instanceof ViewGroup) {
-            final ViewGroup viewGroup = (ViewGroup) view;
+        final Queue<View> viewsToProcess = new LinkedList<>();
+        viewsToProcess.add(view);
+
+        while (!viewsToProcess.isEmpty()) {
+            final View viewToProcess = viewsToProcess.remove();
```

Comment:
Have you tested this? Usually it is illegal to modify a collection while iterating over it and will result in a [ConcurrentModificationException](https://docs.oracle.com/javase/7/docs/api/java/util/ConcurrentModificationException.html) but it's possible I'm wrong. You can get around this by using an [Iterator](https://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html).